### PR TITLE
feat(project_id): checkout project_id exists

### PR DESCRIPTION
**Pull request by thanh-nguyen-dang**
Jira Ticket: [PXP-8820](https://ctds-planx.atlassian.net/browse/PXP-8820)
### New Features
- Check a field called `project_id` exist for an index

### CODE REVIEW

File: gen3utils/etl/etl_validator.py [hunk 1]
```diff
@@ -101,6 +102,7 @@ def validate_prop(
     recorded_errors,
     grouping_path=None,
     nodes_for_category=None,
+    is_root_node=False,
```
Comment (paulineribeyre) [inline gen3utils/etl/etl_validator.py:105]: `is_root_node` is not used

File: gen3utils/etl/etl_validator.py [hunk 2]
```diff
@@ -364,6 +366,13 @@ def check_mapping_constraints(mappings, model, recorded_errors, underscore):
                     index,
                     nodes_for_category,
                 )
+        if "project_id" not in index.props and m.get("type") == "aggregator":
+            recorded_errors.append(
+                PropertiesError(
+                    f"project_id is not found in a root node of {index.name}"
```
Comment (paulineribeyre) [inline gen3utils/etl/etl_validator.py:372]:
```suggestion
                    f"project_id is not found in index {index.name}"
```

File: gen3utils/etl/etl_validator.py [hunk 3]
```diff
@@ -364,6 +366,13 @@ def check_mapping_constraints(mappings, model, recorded_errors, underscore):
                     index,
                     nodes_for_category,
                 )
+        if "project_id" not in index.props and m.get("type") == "aggregator":
```
Comment (paulineribeyre) [inline gen3utils/etl/etl_validator.py:369]: is `project_id` not required for "collector" mappings?

File: gen3utils/manifest/manifest_validator.py [hunk 4]
```diff
@@ -332,7 +328,7 @@ def version_requirement_validation(
 
 def misc_validations(manifest):
     """
-    Handles all the ad-hoc validations which are not structurally grouped in the validation_config.yml
+    Handles all the sad-hoc validations which are not structurally grouped in the validation_config.yml
```
Comment (paulineribeyre) [inline gen3utils/manifest/manifest_validator.py:331]:
```suggestion
    Handles all the ad-hoc validations which are not structurally grouped in the validation_config.yml
```

File: gen3utils/etl/etl_validator.py (hunk 3 above)
Comment (thanh-nguyen-dang) [reply gen3utils/etl/etl_validator.py:369]: I think so, because we can collect some file  which not belonging any project @paulineribeyre

Comment (paulineribeyre) [reply gen3utils/etl/etl_validator.py:369]: oh ok!

**Review by paulineribeyre — APPROVED** [review]

**Pull request merged**